geowrangler/datasets/ookla.py

```diff
@@ -41,14 +68,208 @@ def list_ookla_files() -> dict:
+            return df
+        else:
+            gdf = gpd.read_file(cached_file_path, driver="GeoJSON")
+            self.data_cache[data_key] = gdf
+            return gdf
+
+    logger.debug("No cached data found. Processing Ookla data from scratch.")
+
+    # Otherwise, load from raw file and add to RAM cache
+    type_year_cache_dir = download_ookla_year_data(
+        type_,
+        year,
+        cache_dir=self.cache_dir,
+        use_cache=use_cache,
+    )
+
+    # Generate the bing tile quadkeys that intersect with the input aoi
+    logger.debug(
+        f"Generating bing tile grids for aoi total bounds {np.array2string(aoi.total_bounds, precision=6)}"
+    )
+    bing_tile_grid_generator_no_geom = grids.BingTileGridGenerator(
+        OOKLA_QUADKEY_LEVEL, return_geometry=False
+    )
+    aoi_quadkeys = bing_tile_grid_generator_no_geom.generate_grid(aoi)[
+        "quadkey"
+    ].tolist()
+
+    # Combine quarterly data for the specified year, filtered to the aoi using quadkey
+    # Quarter is inferred from the Ookla filename
+    quarter_df_list = []
+    for ookla_filename in sorted(os.listdir(type_year_cache_dir)):
+        ooklaFile = lookup_ookla_file(ookla_filename)
+        if ooklaFile:
+            ookla_quarter_filepath = os.path.join(type_year_cache_dir, ookla_filename)
+            logger.debug(
+                f"Ookla data for aoi, {type_} {year} {ooklaFile.quarter} being loaded from {ookla_quarter_filepath}"
+            )
+            quarter_df = pd.read_parquet(ookla_quarter_filepath)
+            # TODO optimize filtering via merge
+            quarter_df = quarter_df[quarter_df["quadkey"].isin(aoi_quadkeys)]
+            quarter_df["quarter"] = int(ooklaFile.quarter)
+            quarter_df_list.append(quarter_df)
+
+            # Free memory after processing
+            del quarter_df
+
+    logger.debug(
+        f"Concatenating quarterly Ookla data for {type_} and {year} into one dataframe"
+    )
+    df = pd.concat(quarter_df_list, ignore_index=True)
+
+    # NOTE: Since there will be groupby operations in processing, we don't return
+    #       a geodataframe by default since it does not work well with aggregations
+    #       by quadkey.
+    write_ookla_metajson(
+        self.processed_cache_dir,
+        data_key,
+        aoi.total_bounds,
+        type_,
+        year,
+        return_geometry,
+    )
+
+    if not return_geometry:
+        logger.debug(f"Saving Ookla data into csv file {cached_file_path}")
+        self.data_cache[data_key] = df
+        df.to_csv(cached_file_path, index=False)
+        return df
+    else:
+        logger.debug(f"Converting Ookla data into geojson file {cached_file_path}")
+        gdf = gpd.GeoDataFrame(df, geometry=gpd.GeoSeries.from_wkt(df["tile"]))
+        self.data_cache[data_key] = gdf
+        logger.debug(f"Saving Ookla data into geojson file {cached_file_path}")
+        gdf.to_file(cached_file_path, driver="GeoJSON")
```

Comment:
@jtmiclat  any reason for not preferring GeoJSON?

Reply:
JSON is not a good format for large datasets. The biggest issue is that parsing a json requires the entire file to be parsed because of the nature of json. Ideal formats have some form of partial reads (ie reading some rows)

Reply:
Will consider this as an enhancement (optimization) request to use GeoJSONSeq instead of GeoJSON in the future. 
See [this](https://github.com/thinkingmachines/geowrangler/discussions/202)